File: bubblesort.py
def bubbleSort(arr):
    n = len(arr)
    for i in range(n-1):
        for j in range(0, n-i-1):
            if arr[j+1] < arr[j]:
                tmp = arr[j]
                arr[j] = arr[j+1]
                arr[j+1] = tmp

#Test cases
test1 = [23, 78, 43, 14, 90, 82]
test2 = [65, 32, 17, 75, 55, 4]

bubbleSort(test1)
bubbleSort(test2)

print("Test 1: ")
for i in range(len(test1)):
    print("%d" %test1[i])

print("Test 2: ")
for i in range(len(test2)):
    print("%d" %test2[i])

File: bubblesort.java
class Bubblesort
{
    void bubbleSort(int arr[])
    {
        int n = arr.length;
        for (int i = 0; i < n-1; i++)
        {
            for (int j = 0; j < n-i-1; j++)
            {
                if (arr[j+1] < arr[j])
                {
                    int tmp = arr[j];
                    arr[j] = arr[j+1];
                    arr[j+1] = tmp;
                }
            }
        }
    }

    void printArr(int arr[]) //Simple loop to iterate through the array and print its elements
    {
        for (int i = 0; i < arr.length; i++)
        {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }

    public static void main(String[] args)
    {
        Bubblesort sort = new Bubblesort();
        
        //Test cases
        int test1[] = {23, 78, 43, 14, 90, 82};
        int test2[] = {65, 32, 17, 75, 55, 4};

        sort.bubbleSort(test1);
        sort.bubbleSort(test2);

        System.out.println("Test 1: ");
        sort.printArr(test1);
        System.out.println("Test 2: ");
        sort.printArr(test2);
    }
}